### PR TITLE
refactor(platform/bitbucket-server): split initRepo

### PR DESCRIPTION
**Pull request by jatladams**
## Changes

Refactor bitbucket server `initRepo()` function to split get repo url logic into its own function

## Context

Requested here to make this PR easier to review https://github.com/renovatebot/renovate/pull/13875

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Ran existing tests (changes covered by existing tests)
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/bitbucket-server/index.ts [hunk 1]
```diff
@@ -153,6 +158,41 @@ export async function getJsonFile(
   return JSON.parse(raw);
 }
 
+function getRepoGitUrl(
```
Comment (viceice) [inline lib/modules/platform/bitbucket-server/index.ts:161]: Can you move this to the `./utils.ts` file please

**Review by rarkins — DISMISSED** [review]
Needs utils refactor

File: lib/modules/platform/bitbucket-server/index.ts (hunk 1 above)
Comment (jatladams) [reply lib/modules/platform/bitbucket-server/index.ts:161]: Hi @viceice ,
I have been trying to make this refactor but there seems to be an issue using `import * as git from '../../../util/git';` that I have been unable to resolve. 
When the git import is introduced `yarn test` throws ~270 errors when running `tsc --noEmit -p tsconfig.strict.json` which is beyond my level of understanding

Comment (viceice) [reply lib/modules/platform/bitbucket-server/index.ts:161]: You need to add the utils file to the exclude list in `tsconfig.strict.json`

Comment (jatladams) [reply lib/modules/platform/bitbucket-server/index.ts]: Amazing, thank you for pointing that out! I was not going to solve that myself. Moved `getRepoGitUrl` into utils, let me know if there are any issues 😄

Comment (viceice) [reply lib/modules/platform/bitbucket-server/index.ts]: https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#re-requesting-a-review 😉

**Comment by viceice** [comment]
it seems you used a new gpg key, please link it to your github account😉

**Comment by jatladams** [comment]
Done! 😄

**Review by viceice — APPROVED** [review]

**Pull request merged**